File: servers/jax-rs/src/main/java/com/wordnik/api/StatsApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.client.model.StatsHistory;
import com.wordnik.client.model.Stats;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/stats.json")
@Api(value = "/stats", description = "the stats API")
@Produces({"application/json"})
public class StatsApi {
  @GET
  @Path("/")
  @ApiOperation(value = "Get exchange-wide and per-series turnover and volume statistics.", notes = "", responseClass = "List<Stats>")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response find(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  @GET
  @Path("/history")
  @ApiOperation(value = "Get historical exchange-wide and per-series turnover and volume statistics.", notes = "", responseClass = "List<StatsHistory>")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response history(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  }

File: servers/scalatra/src/main/scala/com/wordnik/client/api/StatsApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.Stats
import com.wordnik.client.model.Error
import com.wordnik.client.model.StatsHistory

import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class StatsApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "StatsApi"
  override protected val applicationName: Option[String] = Some("Stats")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }
  

  val stats.findOperation = (apiOperation[List[Stats]]("stats.find")
      summary "Get exchange-wide and per-series turnover and volume statistics."
      parameters()
  )

  get("/stats",operation(stats.findOperation)) {
    
  }

  

  val stats.historyOperation = (apiOperation[List[StatsHistory]]("stats.history")
      summary "Get historical exchange-wide and per-series turnover and volume statistics."
      parameters()
  )

  get("/stats/history",operation(stats.historyOperation)) {
    
  }

}